aksel.nav.no/website/pages/eksempler/combobox/react-hook-form.tsx

```diff
@@ -0,0 +1,78 @@
+import { Controller, SubmitHandler, useForm } from "react-hook-form";
+import { UNSAFE_Combobox } from "@navikt/ds-react";
+import { withDsExample } from "@/web/examples/withDsExample";
+
+type Inputs = { transportmiddel: string[] };
+
+const Example = () => {
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<Inputs>({
+    reValidateMode: "onBlur",
```

Comment:
Is combobox more aligned with the checkbox pattern of requiring "onChange" as reValidateMod? Since i'm actively making a choice to pick a new value (like "car" here), it seems like something else is wrong if the error still stands

<img width="442" alt="Screenshot 2024-08-29 at 09 46 36" src="https://github.com/user-attachments/assets/f58f7987-472b-442d-b2d0-ce8485dbf8fd">

Reply:
I'm not able to recreate this, when I've selected a value the error disappears for me (firefox).

Reply:
actually, i can get this state if i close the list options by navigating the mouse cursor up one from the first element. So nvm. I guess this would be a confusing state to end up in.

Reply:
Since combobox has a text input field, then the onChange for that should probably not cause rapid changes to the dom / form validation? (we can prevent validation on this onChange, but have it trigger on the selectedOptions onChange (to avoid the above situation).

Reply:
I get this state by only using keyboard or clicking "car" in dropdown then hitting "esc" to close menu. I guess most will click outside the menu to close it thus triggering onBlur, but would be best if it was experienced the same for both mouse and keyboard-users.

Hmm true about the onChange firing raid events, would not be optimal 🤔 Would it be possible to manually update validation-state within `onToggleSelected`? Something like 
```
onToggleSelected: (value, isSelected) => if isSelected -> updateValidation()
```

Reply:
Seems like i didn't properly understand how the `Controller` function worked 🙈 Since we have full control over where `field.onChange` is added, we avoid rapid events like you mentioned Julian 👍 

I tested with this snippet and it seems to work like expected:
- Added `selectedOptions={field.value}` since we have to control the combobox
- Changed to `reValidateMode: "onChange",`

```tsx
const Example = () => {
  const {
    handleSubmit,
    control,
    formState: { errors },
  } = useForm<Inputs>({
    reValidateMode: "onChange",
    defaultValues: { transportmiddel: [] },
  });

  const onValidSubmit: SubmitHandler<Inputs> = (data) => {
    alert(data);
  };

  return (
    <form onSubmit={handleSubmit(onValidSubmit)}>
      <Controller
        control={control}
        rules={{ required: "Du må velge minst ett transportmiddel." }}
        name="transportmiddel"
        render={({ field }) => (
          <UNSAFE_Combobox
            id="transportmiddel"
            label="Hva er de kuleste transportmidlene?"
            options={options}
            isMultiSelect
            error={errors.transportmiddel?.message}
            ref={field.ref}
            name={field.name}
            selectedOptions={field.value}
            onToggleSelected={(option, isSelected) => {
              if (isSelected) {
                field.onChange([...field.value, option]);
              } else {
                field.onChange(field.value.filter((v) => v !== option));
              }
            }}
          />
        )}
      />
    </form>
  );
};
```

Reply:
I agree that we can validate on `onChange` here. "onChange" in this case is actually `onToggleSelected`. The reason for choosing `onBlur` at first was to avoid too early validation if you need to select multiple items. But when thinking about it, that shouldn't be a problem as long as devs don't create too clever error messages.

Let's say you have selected 1 but need to select 3. Error message should be "you need to select at least 3 items". If you remove everyting, the error should be the same. If you add one, the error should still be the same. It would only be a problem if the error said like "select 2 MORE items", but I think that's an edge case. Even then, at least it happens only when selecting, not on every keystroke 😅

Reply:
As far as I know, the teams I have had contact with about Combobox and react-hook-form use `reValidateMode: "onChange"`.